cli/flox-watchdog/src/main.rs

```diff
@@ -64,26 +65,40 @@ pub struct Cli {
     pub disable_metrics: bool,
 }
 
+fn main() -> ExitCode {
+    let args = Cli::parse();
+
+    // Initialization
+    let log_file = &args
+        .log_dir
+        .as_ref()
+        .map(|dir| dir.join(format!("watchdog.{}.log", args.pid)));
+
+    init_logger(log_file)
+        .context("failed to initialize logger")
+        .unwrap();
+    let _sentry_guard = (!args.disable_metrics).then(init_sentry);
+
+    // Main
+    match run(args) {
+        Err(_) => ExitCode::FAILURE,
+        Ok(_) => ExitCode::SUCCESS,
+    }
+}
+
 #[instrument("watchdog",
+    err(Debug),
     skip_all,
-    fields(
-        pid = tracing::field::Empty,
+    fields(pid = tracing::field::Empty,
         registry = tracing::field::Empty,
         dot_flox_hash = tracing::field::Empty,
         socket = tracing::field::Empty,
         log_dir = tracing::field::Empty))]
-fn main() -> Result<(), Error> {
-    // Initialization
-    let args = Cli::parse();
-    let log_file = &args
-        .log_dir
-        .as_ref()
-        .map(|dir| dir.join(format!("watchdog.{}.log", args.pid)));
-    init_logger(log_file).context("failed to initialize logger")?;
+fn run(args: Cli) -> Result<(), Error> {
     if let Err(err) = ensure_process_group_leader() {
         error!(%err, "failed to ensure watchdog is detached from terminal");
     }
```

Comment:
(non-blocking suggestion) Can we just call `ensure_process_group_leader()?` and `wait_for_termination(…)?` (or optionally with `.context(…)`) now that errors are reported correctly?

Reply:
yes, if that was what the `error` meant to do

Reply:
n.b. do you know why `cleanup` is not returning a result?